infra/Pos/Slotting/MemState/Holder.hs

```diff
@@ -42,22 +51,72 @@ class HasSlottingVar ctx where
 type SlotsDefaultEnv ctx m =
     (MonadReader ctx m, HasSlottingVar ctx, MonadIO m)
 
-getSystemStartDefault :: SlotsDefaultEnv ctx m => m Timestamp
+getSystemStartDefault
+    :: SlotsDefaultEnv ctx m
+    => m Timestamp
 getSystemStartDefault = view slottingTimestamp
 
-getSlottingDataDefault :: SlotsDefaultEnv ctx m => m SlottingData
-getSlottingDataDefault = atomically . readTVar =<< view slottingVar
+getAllEpochIndicesDefault
+    :: SlotsDefaultEnv ctx m
+    => m [EpochIndex]
+getAllEpochIndicesDefault = do
+    var <- view slottingVar
+    atomically $ getAllEpochIndices <$> readTVar var
+
+getCurrentEpochIndexDefault
+    :: SlotsDefaultEnv ctx m
+    => m EpochIndex
+getCurrentEpochIndexDefault = do
+    var <- view slottingVar
+    atomically $ getCurrentEpochIndex <$> readTVar var
+
+getNextEpochIndexDefault
+    :: SlotsDefaultEnv ctx m
+    => m EpochIndex
+getNextEpochIndexDefault = do
+    var <- view slottingVar
+    atomically $ getNextEpochIndex <$> readTVar var
+
+getCurrentEpochSlottingDataDefault
+    :: SlotsDefaultEnv ctx m
+    => m EpochSlottingData
+getCurrentEpochSlottingDataDefault = do
+    var <- view slottingVar
+    atomically $ getCurrentEpochSlottingData <$> readTVar var
+
+getNextEpochSlottingDataDefault
+    :: SlotsDefaultEnv ctx m
+    => m EpochSlottingData
+getNextEpochSlottingDataDefault = do
```

Comment:
I meant you used the same pattern of code in these functions. and maybe it makes sense to write a function which will accept getter and do this pattern inside